src/commonMain/kotlin/au/id/micolous/metrodroid/transit/lisboaviva/LisboaVivaSubscription.kt

```diff
@@ -53,11 +55,24 @@ class LisboaVivaSubscription (override val parsed: En1545Parsed,
             val units = parsed.getIntOrZero(CONTRACT_PERIOD_UNITS)
             when (units) {
                 0x109 -> return vf + Duration.daysLocal(period - 1)
-                0x10a -> return vf + Duration.monthsLocal(period - 1)
+                0x10a -> {
+                    // It's calendar months. Hence this trickery
+                    val ymdStart = vf.ymd
+                    val ymStart = ymdStart.year * 12 + ymdStart.month
+                    val ymEnd = ymStart + period
+                    val dEnd = Daystamp(year = ymEnd / 12, month = ymEnd % 12, day = 1)
+                    return dEnd + Duration.daysLocal(-1)
+                }
             }
             return super.validTo
         }
 
+    override fun getAgencyName(isShort: Boolean): FormattedString? {
+        if (contractProvider == 31)
```

Comment:
Is `31` here `ZAPPING_AGENCY`?

Reply:
Not really. It's used for different interagency subscriptions. Renamed